### PR TITLE
Added a method to compute data drift on a ME

### PR DESCRIPTION
**Pull request by fterrazzoni**
Relates to https://github.com/dataiku/dip/pull/13469

### CODE REVIEW

File: dataikuapi/dss/modelevaluationstore.py [hunk 1]
```diff
@@ -288,6 +289,22 @@ def delete(self):
         self.client._perform_json(
                 "DELETE", "/projects/%s/modelevaluationstores/%s/runs/" % (self.project_key, self.mes_id), body=obj)
 
+    def compute_data_drift(self, reference_id=None, data_drift_params=None):
+        """
+        Compute data drift against a reference model or model evaluation. The reference is determined automatically unless specified.
```
Comment (lpenet) [inline dataikuapi/dss/modelevaluationstore.py:294]: It seems to me that computing drift against a lab model is not tested. I would make this limitation clearer there.

**Review by lpenet — CHANGES_REQUESTED** [review]

File: dataikuapi/dss/modelevaluationstore.py (hunk 1 above)
Comment (fterrazzoni) [reply dataikuapi/dss/modelevaluationstore.py:294]: Nothing here specifically mentions "Lab models"? What do you want to change exactly?

(allowing or disallowing this is another debate, to have in the backend PR)

**Review by instanceofme — APPROVED** [review]

File: dataikuapi/dss/modelevaluationstore.py (hunk 1 above)
Comment (lpenet) [reply dataikuapi/dss/modelevaluationstore.py:294]: Yes it was none mentioned here. I just wanted to make it clear. It seems to me that we decided to have some positive wording. Something like "Compute data drift against a reference saved model version or a model evaluation"

**Comment by lpenet** [comment]
Playing with it, I felt quite incomfortable with the API, despite the fact that it works.
The user can not easily get the referenceId of a ME or a SMV through the API.
We discussed this with @fterrazzoni and agreed that it would be nice to also allow to pass a `ModelEvaluation` or a `DSSTrainedPredictionModelDetails` as a parameter.
It would also be nice to allow the user to access the fullId easily. But as this is an API change, I guess your opinion is required, @instanceofme

File: dataikuapi/dss/ml.py [hunk 2]
```diff
@@ -2499,6 +2499,10 @@ def get_roc_curve_data(self):
 
         return roc
 
+    @property
+    def _full_model_like_id(self):
+        return self.details["fullModelId"]
+
```
Comment (lpenet) [inline dataikuapi/dss/ml.py:2505]: I would set that in the base class, as it should work for clustering models too.

Comment (fterrazzoni) [reply dataikuapi/dss/ml.py:2505]: I'm not sure as drift won't work on a clustering model ? 

Or maybe, you wanted this method to be called by end users and in this case, we probably want to make it public ? WDYT ?

Comment (lpenet) [reply dataikuapi/dss/ml.py:2505]: Obviously, it will not be used for drift. :-)

Yes, I think it could be useful to users and we should make it public.

**Review by instanceofme — APPROVED** [review]

File: dataikuapi/dss/ml.py (hunk 2 above)
Comment (fterrazzoni) [reply dataikuapi/dss/ml.py]: Done

File: dataikuapi/dss/modelevaluationstore.py [hunk 3]
```diff
@@ -288,6 +289,31 @@ def delete(self):
...
+        return "ME-%s-%s-%s"%(self.project_key, self.mes_id, self.run_id)
+
+    def compute_data_drift(self, reference=None, data_drift_params=None):
+        """
+        Compute data drift against a reference model or model evaluation. The reference is determined automatically unless specified.
+
+        :param reference: saved model version or model evaluation to use as reference (optional, accepts ID or Python object)
+        :type reference: Union[str, DSSModelEvaluation, DSSTrainedPredictionModelDetails]
+        :param data_drift_params: data drift computation settings (optional)
+        :return: data drift analysis results, as a JSON object
```
Comment (lpenet) [inline dataikuapi/dss/modelevaluationstore.py:303]: Sorry, I have a last remark. A DSSFuture is returned, not a JSON object. I would write something like:
```
:returns: a :class:`dataikuapi.dss.future.DSSFuture` tracking the progress of the update. Call 
                   :meth:`~dataikuapi.dss.future.DSSFuture.wait_for_result` on the returned object
                   to wait for completion (or failure) and get the result as a JSON object
```

**Review by lpenet — CHANGES_REQUESTED** [review]
Almost good. One last remark on the doc. Seems important as a user could get perplex, else.

File: dataikuapi/dss/modelevaluationstore.py (hunk 3 above)
Comment (fterrazzoni) [reply dataikuapi/dss/modelevaluationstore.py]: Handled differently (auto wait by default, much more practical)

**Review by lpenet — APPROVED** [review]
LGTM+tested with Python 2.7 and 3.7. Would like to enrich API doc, but will turn that in an opportunity of learning for a newcomer. :-)

**Pull request merged**